### PR TITLE
feat: hold invoice

### PR DESCRIPTION
**Pull request by nicolasburtey**
this PR make invoice been settled solely by trigger, remove the need to query lnd from a main query from the user.

This is a [big performance optimization](https://ui.honeycomb.io/galoy/datasets/galoy-bbw/result/EGqhArMX7R2).

### CODE REVIEW

**Comment by nicolasburtey** [comment]
we could revisit once this is merged: https://github.com/lightningnetwork/lnd/issues/6338

**Comment by nicolasburtey** [comment]
from Alex:

> I don't think there would be any issue from the LND perspective, although you can of course cancel the single invoice subscription after it's settled or expired/canceled so I would use that to minimize overall subscriptions

maybe we could also limit the BTC lightning invoice to something like 4h instead of the current 24h?

File: src/domain/bitcoin/lightning/index.ts [hunk 1]
```diff
@@ -18,3 +20,16 @@ export const PaymentSendStatus = {
   Pending: { value: "pending" },
   AlreadyPaid: { value: "already_paid" },
 } as const
+
+export const lnPaymentStatusEvent = (paymentHash: PaymentHash) =>
```
Comment (vindard) [inline src/domain/bitcoin/lightning/index.ts:24]: This isn't used anywhere

File: src/graphql/error-map.ts [hunk 2]
```diff
@@ -271,6 +271,10 @@ export const mapError = (error: ApplicationError): CustomApolloError => {
       message = "Invalid walletId for account."
       return new ValidationInternalError({ message, logger: baseLogger })
 
+    case "SecretDoesNotMatchAnyExistingHodlInvoiceError":
+      message = `Unknown error occurred (code: ${error.name})`
```
Comment (vindard) [inline src/graphql/error-map.ts:275]: This message is a duplicate. The case can be added to the unhandled list below Ln 280

**Review by vindard — CHANGES_REQUESTED** [review]
Initial review, 2 small nits

File: src/graphql/error-map.ts (hunk 2 above)
Comment (nicolasburtey) [reply src/graphql/error-map.ts:275]: yup thanks

**Comment by nicolasburtey** [comment]
closing, as this PR has been replaced by #1395

**Pull request closed without merging**